src/common/validators/scripture-references.validator.ts

```diff
@@ -0,0 +1,52 @@
+import { ValidationArguments, ValidationOptions } from 'class-validator';
+import { books } from '../../components/scripture/books';
+import { bookIndexFromName } from '../../components/scripture/reference';
+import { ValidateBy } from './validateBy';
+
+export const IsValidChapter = (
+  book: string,
+  validationOptions?: ValidationOptions
+) =>
+  ValidateBy(
+    {
+      name: 'isValidChapter',
+      constraints: [book],
+      validator: {
+        validate: (value, args: ValidationArguments) => {
+          const [book] = args.constraints;
+          const bookName = (args.object as any)[book];
+          const bookIndex = bookIndexFromName(bookName);
+          return value <= books[bookIndex].chapters.length;
```

Comment:
Could we move the actual validation logic to a `validateChapter(book, chapter)` function in the reference file?
It would be better for separation of concerns. That can handle the actual logic and this file can just convert that function to a validation decorator.

Reply:
Gotcha